ibis/expr/types/core.py

```diff
@@ -48,7 +48,18 @@ def __repr__(self) -> str:
             )
             return output
         else:
-            return repr(result)
+            try:
+                from rich.console import Console
+            except ImportError:
```

Comment:
i would encourage this to be a separate method (maybe called from here)

repr can be called in many places and if doing a lot of work can be a perf issue in graph construction 

eg repr should itself be very simple, but maybe options can control more advanaced things

Reply:
The imports have been moved to toplevel.